simon.py

```diff
@@ -0,0 +1,132 @@
+import time
+import os.path
+import json
+import numpy as np
+import pandas as pd
+from sklearn.metrics import f1_score
+
+import sys
+sys.path.append("./simon")
+from Simon import Simon
+from Simon.Encoder import Encoder
+from Simon.DataGenerator import DataGenerator
+
+from d3m_profiler.build_table import get_datasets
+
+MAX_LEN = 20
+MAX_CELLS = 100
+P_THRESHOLD = 0.3
+CHECKPOINT_DIR = 'simon/Simon/pretrained_models/'
+DATASET_DIR = '/users/data/d3m/datasets'
+
+def main(checkpoint, nb_epoch, batch_size, execution_config):
+    if not os.path.isdir(CHECKPOINT_DIR):
+        os.makedirs(CHECKPOINT_DIR)
+
+    raw_data, header, categories = parse_datasets(get_datasets(DATASET_DIR))
+    print(f'Categories: {categories}', flush=True)
+
+    start = time.time()
+
+    encoder = Encoder(categories=categories)
+    encoder.process(raw_data, MAX_CELLS)
+    X, y = encoder.encode_data(raw_data, header, MAX_LEN)
+    encoder_max_cells = encoder.cur_max_cells
+    category_count = y.shape[1] # need to know number of fixed categories to create model
+    print('Number of fixed categories is: {}'.format(category_count))
+
+    classifier = Simon(encoder=encoder)
+    data = classifier.setup_test_sets(X, y)
+    model = classifier.generate_model(MAX_LEN, encoder_max_cells, category_count)
+    classifier.load_weights(checkpoint, {}, model, CHECKPOINT_DIR)
+    model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['binary_accuracy'])
+    history = classifier.train_model(batch_size, CHECKPOINT_DIR, model, nb_epoch, data)
+
+    config = { 'encoder' :  encoder,
+               'checkpoint' : classifier.get_best_checkpoint(CHECKPOINT_DIR) }
+    classifier.save_config(config, CHECKPOINT_DIR)
+    classifier.plot_loss(history) # comment out on docker images...
+
+    evaluate_model(model, data)
+    print(f'Total time taken: {time.time()-start}')
+
+
+def evaluate_model(model, data):
+    start = time.time()
+    scores = model.evaluate(data.X_test, data.y_test, verbose=0)
+    end = time.time()
+    print("Accuracy: %.2f%% \n Time: {0}s \n Time/example : {1}s/ex".format(
+        end - start, (end - start) / data.X_test.shape[0]) % (scores[1] * 100))
+    
+    probabilities = model.predict(data.X_test, verbose=1)
+    prediction_indices = probabilities > P_THRESHOLD
+    y_pred = np.zeros(data.y_test.shape)
+    y_pred[prediction_indices] = 1
+    print(f'F1 micro: {f1_score(data.y_test, y_pred, average="micro")}')
+    print(f'F1 macro: {f1_score(data.y_test, y_pred, average="macro")}')
+    print(f'F1 weighted: {f1_score(data.y_test, y_pred, average="weighted")}')
+
+
+def parse_datasets(datasets):
```

Comment:
d3m has functionality for loading d3m datasets: https://gitlab.com/datadrivendiscovery/d3m/-/blob/devel/d3m/container/dataset.py#L3193

Using this means we don't have to maintain our own dataset parsing code.